src/script/sign.js

```diff
@@ -0,0 +1,83 @@
+const email_input = document.querySelector("#signin_email");
+const password_input = document.querySelector("#signin_password");
+const error_email = document.querySelector("#error_email");
+const error_password = document.querySelector("#error_password");
+const eyes = document.querySelector(".eye_button");
+
+// 이메일 검사
+function validateEmail() {
+    // 이메일 값이 없는 경우
+    if (email_input.value.trim() === "") {
+        email_input.style.border = "1px solid var(--Linkbrary-red)";
+        error_email.innerText = "이메일을 입력해 주세요.";
+        error_email.style.display = "block";
+    }
+    // 올바른 이메일 주소가 아닐 경우
+    else if (!isValidEmail(email_input.value)) {
+        email_input.style.border = "1px solid var(--Linkbrary-red)";
+        error_email.innerText = "올바른 이메일 주소가 아닙니다.";
+        error_email.style.display = "block";
+    }
+    // 이메일 값이 있는 경우
+    else {
+        email_input.style.border = "1px solid var(--Linkbrary-gray20)";
+        error_email.style.display = "none";
+    }
```

Comment:
(제취향) early return을 이용해 조금더 가독성을 높히는 방법도 있습니다.

```suggestion
    // 이메일 값이 없는 경우
    if (email_input.value.trim() === "") {
        email_input.style.border = "1px solid var(--Linkbrary-red)";
        error_email.innerText = "이메일을 입력해 주세요.";
        error_email.style.display = "block";
        return;
    }
    // 올바른 이메일 주소가 아닐 경우
    if (!isValidEmail(email_input.value)) {
        email_input.style.border = "1px solid var(--Linkbrary-red)";
        error_email.innerText = "올바른 이메일 주소가 아닙니다.";
        error_email.style.display = "block";
        return;
    }
    // 이메일 값이 있는 경우
      email_input.style.border = "1px solid var(--Linkbrary-gray20)";
      error_email.style.display = "none";
```

Reply:
* early return은 과연 좋은가?
https://thearchivelog.dev/article/are-early-returns-any-good/